Enable log compression for pipeline jobs

maniaczek opened the pull request:
When using pipeline style job it is possible to generate huge logs. It is easy to compress this with this small change.
This change is to compress composite log only. To compress single step log it requires more complex changes most probably outside of this plugin.

ArturHarasimiuk committed:
Enable log compression for pipeline jobs

oleg-nenashev commented:
Not sure it can work in such case, needs testing to the codebase review.
Some tests could definitely help

ArturHarasimiuk committed:
add test for isApplicable on WorkflowJob

Create Pipeline project and pass its .class to make sure that
isApplicable returns true.

oleg-nenashev reviewed (approved):
It should work from what I see

daniel-beck commented:
Superseded by #5.
